### ISSUE TITLE
Server side query to populate list of select2 options

### ISSUE THREAD

**Issue by kumaravnish** [vedmack/yadcf#160]
First of all let me say that this is a very helpful plugin!!!

I have one small problem using it in my project though. My implementation requires me to populate the list of options in the select2 dropdown using a server side query as the table is paginated and does not contain all the options that the user can select from. Moreover, the list of options in the select2 dropdown from the server side can be large, so we want to allow a "type at least 3 letters, before the server request is sent", option which is supported by select2 (query option). However, when I passed in this option to select_type_options object, I got an error in the browser saying:

> "Option 'query' is not allowed for Select2 when attached to a <select> element."

Reading the select2 documentation I realized that:

> "In order for this function to work Select2 should be attached to a input type='hidden' tag instead of a select."

Is there a way to make this work as an option to the select_type="select2"? Or is there any other way you think I will be able to achieve this requirement using your plugin? I hate to have this plugin rejected for our project just for this problem.

**Comment by vedmack** [comment]
In order to populate the select inputs (or any other yadcf input) you should send the relevant data from your server in `yadcf_data_XXX` where XXX stands for column number, if you will inspect the network tab of the [server side showcase page](http://yadcf-showcase.appspot.com/server_side_source.html) you will see something like this:

```
aaData: [["Trident", "Internet Explorer 4.0", "Win 95+", "1/31/2015", "1"],…]
draw: "1"
recordsFiltered: 10
recordsTotal: 57
yadcf_data_0: ["KHTML", "Webkit", "Trident", "Misc", "Other browsers", "Tasman", "Presto", "Gecko"]
yadcf_data_1: ["Nintendo DS browser", "Netscape Browser 8", "All others", "Lynx", "Mozilla 1.6", "Mozilla 1.5",…]
yadcf_data_2: ["N800", "Win 95+ / Mac OS 8.6-9.2", "S60", "KDE 3.5", "Win XP SP2+", "KDE 3.3", "OSX.3", "KDE 3.1",…]
yadcf_data_4: ["134", "0"]
```


As to the "type at least 3 letters" I would suggest something better, use the [filter_delay](https://github.com/vedmack/yadcf/blob/master/jquery.dataTables.yadcf.js#L177) option just like I do in the showcase

**Comment by kumaravnish** [comment]
Thank you for the quick response.

The problem with this approach is that the amount of data being transferred from the server as part of the original request is too large. The list in yadcf_data_xxx could be a list of 5000 users from an ldap server. Hence we would like to use the select2 dropdowns query option which sends a query to the server as the user types something to filter on, instead of prepopulating the list with all the users in the system. 

Similarly, the filter_delay is useful to delay the filtering action in the table. However, here we are talking about filter the data in the dropdown itself and not the table.

Does that make sense? or have I confused you more? :-)

**Comment by vedmack** [comment]
You got that error ("Option 'query' is not allowed for Select2 when attached to a element.") because you havent used the full version of select2, [grab the js here](https://github.com/select2/select2/blob/master/dist/js/select2.full.min.js) and replace your currently included version, (do the same for css file too) then you can add something like this to your code

```
	var obj = {};
	obj.query = function (query) {
		if (query.term && query.term.length  > 2) {
			$.ajax({
				url: "select2_example_query?q=" + query.term
			}).done(function (data) {
				query.callback({ results: data.myData });
			});
		}
	};
```

and tell yadcf to use it, like this `select_type_options: obj`

you server should eventually return something like this

```
"{\"myData\":[{\"id\":\"aaa\",\"text\":\"AAA\"},{\"id\":\"bbb\",\"text\":\"BBB\"}]}";
```

I'm not sure how to "listen" the select event (when selecting something from the returned values.

So in order for me to be able to help you on this one - you should do some reading and when you finish and have some more data for me let me know so I can do the required fixes, basically I need to see a working example of such feature.

Note that 
>Select2: The `query` option has been deprecated in favor of a custom data adapter that overrides the `query` method. Support will be removed for the `query` option in future versions of Select2.

bottom line is that I need to see a working example...

**Comment by kumaravnish** [comment]
Unfortunately I cannot share the project I am working on. I will create some sample code for you to look at. Thanks...

**Comment by vedmack** [comment]
Closing this one (will reopen it when a *working example*  will be available)
